### PR TITLE
fix(linux): Remove version.sh and get tier/version from .md files

### PR DESCRIPTION
**Pull request by darcywong00**
We're revamping our CI and removing our uses of environment variables for tier and version.

atm, I'm trying to see if the reconf.sh script can use the corresponding .md files and remove `version.sh`
Will need to see if the Jenkins CI is affected...

* Also changes `kmconfig --version` to return VERSIONWITHTAG (-alpha, etc)
* CI build also includes the version in the tarball `keyman-config-${VERSION}.tar.gz` (for a few months, the version was not in the name)

### CODE REVIEW

File: linux/scripts/dist.sh
Comment (darcywong00) [inline linux/scripts/dist.sh:80]: Before this PR gets merged, we'll need to update the Keyman - Linux Release-14.0 config

There's a TODO on step 6 (l.23)

**Comment by darcywong00** [comment]
@ermshiperete carried this PR over the finish line. His changes lgtm, but I can't approve this PR since I created it :)

**Review by mcdurdin — APPROVED** [review]
LGTM, don't forget to update CI

**Comment by darcywong00** [comment]
> don't forget to update CI

And done.

**Pull request merged**
